common/app/templates/inlineJS/blocking/applyAdfreeRenderCondition.scala.js

```diff
@@ -0,0 +1,119 @@
+@()
+
+(function applyAdfreeRenderCondition(commercial){
+    var AD_FREE_COOKIE = 'gu_adfree_test',
+        MVT_ID_COOKIE = 'GU_mvt_id',
+        FIRST_TIME_VISIT_KEY = 'gu.firstVisitTime',
+        ALREADY_VISITED_KEY = 'gu.alreadyVisited';
+
+    if ((readLocalStorage(ALREADY_VISITED_KEY) || 0) == 0) {
+        writeLocalStorage(FIRST_TIME_VISIT_KEY, new Date().getTime());
+    }
+
+    if (isFirst3VisitDays() && isInVariantGroup()){
+        commercial.showingAdfree = true;
+        return 'is-adfree';
+    } else {
+        commercial.showingAdfree = false;
+        return '';
+    }
+
+    function isFirst3VisitDays() {
+        var firstVisit = readLocalStorage('gu.firstVisitTime') || 0;
+        return (new Date().getTime() - firstVisit) < 259200000;
+    }
+
+    function isInVariantGroup() {
+        var cookie = readCookie(AD_FREE_COOKIE);
+        if (cookie == null) {
+            assignUser();
+        }
+        return readCookie(AD_FREE_COOKIE) == "variant";
+    }
+
+    function assignUser() {
+        var mvtNumValues = 899999,
+            audience = 0.1, // keep this in sync with new-user-adverts-disabled.js
+            audienceOffset = 0, // keep this in sync with new-user-adverts-disabled.js
+            smallestTestId = mvtNumValues * audienceOffset,
+            largestTestId  = smallestTestId + mvtNumValues * audience,
+            mvtCookieId = readCookie(MVT_ID_COOKIE),
+            variants, testVariantIndex;
+
+        if (mvtCookieId && mvtCookieId > smallestTestId && mvtCookieId <= largestTestId) {
+            variants = [{
+                id: 'variant',
+                test: function () {
+                    writeCookie(AD_FREE_COOKIE, 'variant');
+                }
+            }, {
+                id: 'control',
+                test: function () {
+                    writeCookie(AD_FREE_COOKIE, 'control');
+                }
+            }];
+            testVariantIndex = mvtCookieId % variants.length;
+            variants[testVariantIndex].test();
+        } else {
+            writeCookie(AD_FREE_COOKIE, 'notintest');
+        }
+    }
+
+    function readLocalStorage(key) {
+        if ("localStorage" in window) {
```

Comment:
Although, maybe you've done this so you don't have to pull a util into blocking JS :)